File: problems/python_solutions/contains_duplicate.py
from typing import List


class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        occured: set[int] = {nums[0]}
        for n in nums[1:]:
            if n in occured:
                return True
            occured.add(n)
        return False
         

File: problems/java_solutions/ContainsDuplicate.java
package problems.solutions;

import java.util.HashSet;
import java.util.Set;

public class ContainsDuplicate {
  public boolean hasDuplicate(int[] nums) {
    Set<Integer> occuredNums = new HashSet<>();

    for (int i = 0; i < nums.length; i++) {
      if (occuredNums.contains(nums[i]) == true) {
        return true;
      }
      occuredNums.add(nums[i]);
    }
    return false;

  }
}
